Plugins/PluginsShared/PluginError.swift

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+enum PluginError: Error {
+  // Build plugin
+  case incompatibleTarget(String)
+  case noConfigurationFilesFound
+}
+
+extension PluginError: CustomStringConvertible {
+  var description: String {
+    switch self {
+    case .incompatibleTarget(let string):
+      "Build plugin applied to incompatible target."
+    case .noConfigurationFilesFound:
+      "No configuration files found."
```

Comment:
Can you expand on this to say that there needs to be a file called "blah" somewhere?

Reply:
This is no longer needed due to refactoring